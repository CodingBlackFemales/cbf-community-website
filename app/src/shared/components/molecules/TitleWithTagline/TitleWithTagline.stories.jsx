import React from 'react';

import { storiesOf } from '@storybook/react';

import TitleWithTagline from '.';

storiesOf('Molecules', module).add('Title With Tagline', () => (
  <TitleWithTagline titleText={'Blog'} taglineText={'A tagline'} />
));
