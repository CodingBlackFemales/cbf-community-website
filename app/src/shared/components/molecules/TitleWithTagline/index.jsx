import React from 'react';
import styled from 'styled-components';
import Title from '../../atoms/Title';
import Tagline from '../../atoms/Tagline';

const TitleWithTagline = ({ titleText, taglineText, className }) => {
  const classPrefix = 'title-with-tagline';
  return (
    <div className={(classPrefix, className)}>
      <header className={`${classPrefix}__header`}>
        <Title text={titleText} />
        {taglineText && <Tagline text={taglineText} />}
      </header>
    </div>
  );
};

const StyledTitleWithTagline = styled(TitleWithTagline)`
  text-align: center;

  .title-with-tagline__header {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
  }
`;

export default StyledTitleWithTagline;
