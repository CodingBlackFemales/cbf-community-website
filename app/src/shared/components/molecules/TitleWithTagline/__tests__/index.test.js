import React from 'react';
import { cleanup } from '@testing-library/react';

import TitleWithTagline from '..';
import {
  mountWithTheme,
  renderWithTheme,
} from '../../../../utils/test/testHelpers';

describe('<TitleWithTagline', () => {
  afterEach(cleanup);
  const titleWithTagline = (
    <TitleWithTagline titleText={'A title'} taglineText={'A tagline'} />
  );
  const titleNoTagline = <TitleWithTagline titleText={'A title'} />;

  it('matches the snapshot, when titleText and taglineText exist', () => {
    const { asFragment } = renderWithTheme(titleWithTagline);
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches the snapshot, when titleText exists and taglineText does not exist', () => {
    const { asFragment } = renderWithTheme(titleNoTagline);
    expect(asFragment()).toMatchSnapshot();
  });

  it('displays the title and tagline text', () => {
    const component = mountWithTheme(titleWithTagline);

    const title = component.find('Title');
    const tagline = component.find('Tagline');

    expect(title.exists()).toBe(true);
    expect(tagline.exists()).toBe(true);
  });

  it('displays the title text only', () => {
    const component = mountWithTheme(titleNoTagline);

    const title = component.find('Title');
    const tagline = component.find('Tagline');

    expect(title.exists()).toBe(true);
    expect(tagline.exists()).toBe(false);
  });
});
