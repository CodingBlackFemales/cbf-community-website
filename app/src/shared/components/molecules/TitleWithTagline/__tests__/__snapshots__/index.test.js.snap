// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TitleWithTagline matches the snapshot, when titleText and taglineText exist 1`] = `
<DocumentFragment>
  .c1 {
  text-align: center;
}

.c1 .title__text {
  text-transform: uppercase;
  font-weight: 300;
  margin-bottom: 0.2em;
}

.c2 {
  text-align: center;
}

.c2 .tagline__text {
  margin-top: 0.2em;
  font-weight: 400;
}

.c0 {
  text-align: center;
}

.c0 .title-with-tagline__header {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div
    class="c0"
  >
    <header
      class="title-with-tagline__header"
    >
      <div
        class="c1"
      >
        <h1
          class="title__text"
        >
          A title
        </h1>
      </div>
      <div
        class="c2"
      >
        <p
          class="tagline__text"
        >
          A tagline
        </p>
      </div>
    </header>
  </div>
</DocumentFragment>
`;

exports[`<TitleWithTagline matches the snapshot, when titleText exists and taglineText does not exist 1`] = `
<DocumentFragment>
  .c1 {
  text-align: center;
}

.c1 .title__text {
  text-transform: uppercase;
  font-weight: 300;
  margin-bottom: 0.2em;
}

.c0 {
  text-align: center;
}

.c0 .title-with-tagline__header {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div
    class="c0"
  >
    <header
      class="title-with-tagline__header"
    >
      <div
        class="c1"
      >
        <h1
          class="title__text"
        >
          A title
        </h1>
      </div>
    </header>
  </div>
</DocumentFragment>
`;
